rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -634,10 +634,32 @@ void RSocketStateMachine::handleStreamFrame(
         return;
       }
       VLOG(3) << mode_ << " In: " << framePayload;
-      stateMachine->handlePayload(
-          std::move(framePayload.payload_),
-          framePayload.header_.flagsComplete(),
-          framePayload.header_.flagsNext());
+
```

Comment:
The `StreamStateMachine` has no concept of a fragmented payload though; the only thing that knows about fragments and how to assemble them is the `RSocketStateMachine`. Perhaps the value type of the `streams_` map in `StreamState` should be a pair of `std::pair<StreamFragmentAccumulator, std::shared_ptr<StreamStateMachineBase>>` rather than just a `StreamStateMachineBase`.

Reply:
the PAYLOAD frame has streamId so we always know to which stream it belongs to. Maybe if we would add add fragmentAccumulator to StreamStateMachineBase we would be able to move this logic out of RSStateMachine? (i am just thinking if this class isn't becoming "class everything" with many responsibilities which are impossible to test individually)